spec/formatting.md

```diff
@@ -260,9 +260,20 @@ the following steps are taken:
 
 3. Perform _option resolution_.
 
-4. Call the _function handler_ with the following arguments:
+4. Determine the **_<dfn>function context</dfn>_** for calling the function implementation.
+   This includes:
 
-   - The current _locale_.
+   - The current _locale_,
+     potentially including a fallback chain of locales.
+   - The base directionality of the _message_ and its _text_ tokens.
+
+   If the resolved mapping of _options_ includes any _`u:` options_
+   supported by the implementation, process them as specified.
+   Such `u:` options MAY be removed from the resolved mapping of _options_.
```

Comment:
I think we should be more disciplined about defining terms.

```suggestion
4. Determine the _function context_ for calling the function implementation.
   
   The **_<dfn>function context</dfn>_** contains the context necessary for
   the _function handler_ to perform formatting or selection. It includes:
   - The current _locale_,
     potentially including a fallback chain of locales or locale identifiers.
   - The base directionality of the _message_ and its _text_ tokens.

   If the resolve mapping of _options_ includes any _`u:` options_
   supported by the implementation, process them as specified.
   Such `u:` options MAY be removed from the resolve mapping of _options_.
```

Question: should "function implementation" be "function handler"?

Reply:
> Question: should "function implementation" be "function handler"?

Yes, this text is from before the latter term was introduced.

> The **_<dfn>function context</dfn>_** contains the context necessary for
> the _function handler_ to perform formatting or selection.

This bit should perhaps be reworded a bit, as with our current language technically the _function handler_ doesn't do the formatting or selection, but returns a _resolved value_, which then either does or enables formatting and/or selection to happen.

> the resolve mapping of _options_

Is the suggestion to not say "resolved mapping" intentional? That phrasing seems a bit awkward.

Reply:
> Is the suggestion to not say "resolved mapping" intentional? That phrasing seems a bit awkward.

No, that's a typo. Apparently, I'm very consistent about that specific mistake. Not sure if we should use the term _resolved options_ instead of "resolved mapping of _options_"